File: runners/kre-py/src/context_measurement.py
from datetime import datetime

from influxdb_client import InfluxDBClient, Point, WritePrecision
from influxdb_client.client.write_api import SYNCHRONOUS


# We are using influxdb-client-python that is compatible with 1.8+ versions:
# https://github.com/influxdata/influxdb-client-python#influxdb-1-8-api-compatibility
INFLUX_ORG = ""  # not used
INFLUX_BUCKET = "kre"  # kre is the created database during the deployment
INFLUX_TOKEN = ""  # we don't need authentication


class ContextMeasurement:
    def __init__(self, config, logger):
        self.__config__ = config
        self.__logger__ = logger

        client = InfluxDBClient(url=self.__config__.influx_uri, token=INFLUX_TOKEN)
        self.__write_api__ = client.write_api(write_options=SYNCHRONOUS)

    def save(self, measurement: str, fields: dict, tags: dict):
        point = Point(measurement)

        for key in fields:
            point.field(key, fields[key])

        for key in tags:
            point.tag(key, tags[key])

        point.time(datetime.utcnow(), WritePrecision.NS)

        self.__write_api__.write(INFLUX_BUCKET, INFLUX_ORG, point)

File: runners/kre-go/context_measurement.go
package kre

import (
	"github.com/influxdata/influxdb-client-go"
	"github.com/influxdata/influxdb-client-go/api"
	"github.com/konstellation-io/kre/libs/simplelogger"
	"github.com/konstellation-io/kre/runners/kre-go/config"
	"time"
)

// We are using influxdb-client-go that is compatible with 1.8+ versions:
// https://github.com/influxdata/influxdb-client-go#influxdb-18-api-compatibility
//   - Use the form username:password for an authentication token.
//   - The organization parameter is not used. Use an empty string ("") where necessary.
//   - Use the form database/retention-policy where a bucket is required. Skip retention policy if the default retention policy should be used.
const (
	org    = ""    // not used
	bucket = "kre" // kre is the created database during the deployment
	token  = ""    // we don't need authentication
)

type contextMeasurement struct {
	cfg      config.Config
	logger   *simplelogger.SimpleLogger
	writeAPI api.WriteAPI
}

func NewContextMeasurement(cfg config.Config, logger *simplelogger.SimpleLogger) *contextMeasurement {
	influxCli := influxdb2.NewClient(cfg.InfluxDB.URI, token)
	writeAPI := influxCli.WriteAPI(org, bucket)

	return &contextMeasurement{
		cfg,
		logger,
		writeAPI,
	}
}

func (c *contextMeasurement) Save(measurement string, fields map[string]interface{}, tags map[string]string) {
	p := influxdb2.NewPointWithMeasurement(measurement)

	for f, v := range fields {
		p.AddField(f, v)
	}

	for t, v := range tags {
		p.AddTag(t, v)
	}

	p.SetTime(time.Now())

	c.writeAPI.WritePoint(p)
	c.writeAPI.Flush()
}
